### ISSUE TITLE
Linkdrop, Create Account and Sign In to App - All in One Flow

### ISSUE THREAD

**Issue by mattlockyer** [near/near-wallet#1983]
## Problem

If a user is given a linkdrop to create a new account, they can go through this flow and be redirect to the app, but they are NOT signed in to the app.

This means the user would have to return to the NEAR Wallet and approve the sign in to the app.

Too many steps, when we can do it all in one flow!

## Story

![image](https://user-images.githubusercontent.com/321340/127526420-b7dbf457-8ee9-4519-84b2-876dfc339815.png)

From a screen similar to the above:
1. user clicks "Create Wallet" in some app -> app creates linkdrop
2. [WITH FLAG](https://github.com/near/near-wallet/issues/1982) - skip straight to account creation flow with linkdrop
3. complete account creation
4. WITH FLAG (this issue) - add access key for contractId and methods, this mimics login flow that typically originates in NAJ via wallet.signIn(...)
5. user is redirected to app with access key secret in URL
6. NAJ has option to create ConnectedWalletAccount using access key secret that will update the apps localStorage
7. user refreshes window or closes and comes back - they are still logged into the app

TLDR/WHY; app can linkdrop a user with no crypto, send them to NEAR Wallet onboarding, and have them return fully logged in with an access key.

All in one flow!

## Acceptance Criteria

- [ ] Flag for wallet to add additional access key for app contractId and methodNames to newly created account
- [ ] Wallet redirects to app with access key secret
- [ ] NAJ has ability for app to call new ConnectedWalletAccount or some method e.g. `.fromSecret(...)` and this creates a connectedWalletAccount instance and updates localStorage
- [ ] User is fully logged into app

## NOTES

- Not time sensitive, but can see this becoming a useful pattern

**Comment by MaximusHaximus** [comment]
This sounds like a really smooth flow :+1: I like it!

Related to this PR is our discussion on adding `successUrl` and `failureUrl` URL parameter support to the linkdrop flow.  Those 2 url params are already supported in the login flow.  If we add `failureUrl` to the linkdrop flow, then we take the user through both the account creation flow _and_ the login flows as part of a single visit to the wallet, then error handling will become more complex in the `create wallet` case -- basically, the user  could be redirected to the `failureUrl` due to errors during either step of the process, so they could end up back in your app's `failureUrl` _even though they created an account successfully_, but not have an access key added for the app in that case because _that_ transaction failed for whatever reason.

Do you envision that the app would check on-chain state whenever someone is redirected back to the apps `failureUrl` in order to identify whether the failure was during the account creation process or the login process?  

Or maybe we could add an URL parameter to the failureUrl similar to the `errorCode` parameter that we append to the `failureUrl` when redirecting users from the `/sign` route?

**Comment by Patrick1904** [comment]
@mattlockyer I agree with @MaximusHaximus concerns above. Would the flow outlined below work?
1. App sends user to create account
2. User gets redirected back to apps provided `successUrl`
3. When user lands on the apps `successUrl` - the app requests sign in and send the user to the Wallet `/login` route.

Thinking that this keeps the app 'login' consistent/explicit for all users and gives the app the option to do something once the user has successfully created the account, and it separates the potential errors mentioned by Daryl. There's one back-and-forth between the app and the Wallet, however it's only the first time when the user creates the account.

**Comment by mattlockyer** [comment]
I'm trying to avoid the user being in this "death spiral" of redirects for UX reasons.

It would be optimal if we could send them to /create or /linkdrop with a flag that adds an access key.
Then redirected to successUrl with or without an access key secret.
Currently, I don't need NAJ to help with this, I can just add it myself in memory and localStorage, so this is GTG but something we can add to NAJ for convenience later.

@MaximusHaximus 
The app will always know that the access key wasn't added if the secret isn't present:
- failureUrl - no user account (failed first step)
- successUrl with no access key secret in successUrl (created account, failed login)
- successUrl with access key secret (passed both steps)

I would think about it like this:
- primary action is "create account" and return to app with accountId
- secondary action is "please give this user an access key and return the secret to me"

So it's fine to redirect to successUrl for primary action, but say the secondary action failed and the required access key secret url param is missing.

I share the concern about failure states, but the additional redirect and user click button to rubber stamp something we know they want to do is more of an annoyance for many users in the long run.
